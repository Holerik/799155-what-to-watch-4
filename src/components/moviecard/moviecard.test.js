// moviecard.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Moviecard from '../moviecard/moviecard.jsx';

const filmInfo = {
  id: 1,
  title: `Joker`,
  poster: `img/joker.jpg`,
  altPoster: `Joker poster`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

describe(`Moviecard tests`, () => {
  it(`Moviecard should render poster`, () => {
    const onMovieTitleClick = jest.fn();
    const onMovieCardActivate = jest.fn();
    const onMovieCardOut = jest.fn();
    const tree = renderer
    .create(
        <Moviecard
          movie={filmInfo}
          onMovieCardActivate={onMovieCardActivate}
          onMovieCardOut={onMovieCardOut}
          onMovieTitleClick={onMovieTitleClick}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

