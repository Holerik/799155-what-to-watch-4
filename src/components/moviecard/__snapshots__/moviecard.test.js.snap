// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moviecard tests Moviecard should render poster 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
    id="1"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <a
      href=""
      onClick={[Function]}
    >
      <img
        alt="Joker poster"
        height="175"
        src="img/joker.jpg"
        width="280"
      />
    </a>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href=""
      onClick={[Function]}
    >
      Joker
    </a>
  </h3>
</article>
`;
